### PR TITLE
feat(ios): remove  ios.none device type

### PR DESCRIPTION
**Pull request by noomorph**
## Description

Resolves https://github.com/wix/Detox/issues/3349 .

The new debugging documentation has been covered in https://github.com/wix/Detox/pull/3326 .

> _For API changes:_
 - [x] I have made the necessary changes in the [types index](https://github.com/wix/Detox/blob/master/detox/index.d.ts) file.

### CODE REVIEW

File: detox/src/devices/runtime/drivers/ios/IosDriver.js [hunk 1]
```diff
@@ -1,41 +0,0 @@
-// @ts-nocheck
```
Comment (d4vidi) [inline detox/src/devices/runtime/drivers/ios/IosDriver.js:1]: Well I'm not strongly against that but I must in fact vote for restoring this class

File: examples/demo-react-native/detox.config.js [hunk 2]
```diff
@@ -50,9 +50,14 @@ module.exports = {
       "device": "simulator",
       "app": "ios.debug"
     },
-    "ios.none": {
-      "type": "ios.none",
+    "ios.manual": {
```
Comment (d4vidi) [inline examples/demo-react-native/detox.config.js:53]: Noice!

File: detox/src/environmentFactory.js [hunk 3]
```diff
@@ -85,14 +85,6 @@ function _getFactoryClasses(deviceConfig) {
       runtimeDeviceFactoryClass = runtimeDeviceFactories.IosSimulator;
       break;
 
-    case 'ios.none':
-      envValidatorFactoryClass = envValidationFactories.Noop;
```
Comment (d4vidi) [inline detox/src/environmentFactory.js:89]: Need to also delete the class itself https://github.com/wix/Detox/blob/dbe56701004bc5bc8ac8c490de113616ee1b1336/detox/src/validation/factories/index.js#L25

File: detox/src/devices/runtime/drivers/ios/IosDriver.js (hunk 1 above)
Comment (noomorph) [reply detox/src/devices/runtime/drivers/ios/IosDriver.js:1]: Resolved in bea03863

File: detox/src/environmentFactory.js (hunk 3 above)
Comment (noomorph) [reply detox/src/environmentFactory.js:89]: Disagree! It is reused by ExternalValidator and by Android Emulator.

**Comment by noomorph** [comment]
@d4vidi WDYM? 

> There's actually one more important thing to handle here: https://wix.github.io/Detox/docs/guide/debugging-in-xcode/#add-an-iosnone-configuration-to-detox-section

It is not relevant – please see:

https://github.com/wix/Detox/blob/next/docs/Guide.DebuggingInXcode.md

**Pull request merged**